exonum/src/runtime/mod.rs

```diff
@@ -410,6 +453,17 @@ impl Caller {
             None
         }
     }
+
+    /// Verify that the caller of this method is supervisor service.
```

Comment:
Do we have an explanation in the documentation that only supervisor service has `0` identifier? And thus supervisor is the special kind of services.
Also marker trait for supervisor from #1500 is better idea imo.

Reply:
The problem with any client-enforced method (like forcing the service to implement a marker trait) is that it's trivial to circumvent and therefore difficult to reason about. I'm for this method provided that the dispatcher knows the correspondence between services and implemented interfaces (which seems a good idea in general) and has a marker interface for dispatcher (more generally, for privileged services).